unittests/runtime-strings/bytestest.cpp

```
@@ -171,6 +172,19 @@ BOOST_AUTO_TEST_CASE(bytes2string) {
   BOOST_CHECK_EQUAL(0, memcmp(_1234->data, "1234", 4));
 }
 
+BOOST_AUTO_TEST_CASE(bytes2hexstring) {
+  auto empty = make_string("");
+  auto res = hook_BYTES_bytes2hexstring(empty);
+  BOOST_CHECK(res != empty);
+  BOOST_CHECK_EQUAL(len(res), 0);
+
+  auto _00ff = make_string("\x00\xff", 2);
```

This is not a great test input, as it will mask the error where characters for a given byte are in the wrong order in the output. Maybe `"\x01\xef"`?

---

Also, consider adding a test case for other boundary values, in particular the empty string.

---

Good catch! I improved the test case. The empty string is already covered.